openssl/src/ssl/mod.rs

```diff
@@ -652,6 +654,17 @@ impl SslContextBuilder {
         }
     }
 
+    /// Sets a custom X509Store for verifying peer certificates
+    #[cfg(any(all(feature = "v102", ossl102), all(feature = "v110", ossl110)))]
+    pub fn set_verify_cert_store(&mut self, cert_store: X509Store) -> Result<(), ErrorStack> {
+        unsafe {
+            // set0 will free, set1 increments, and then requires a free
+            let ptr = cert_store.as_ptr();
+            mem::forget(cert_store);
```

Comment:
It doesn't matter in this case since the implementation can't actually fail (https://github.com/openssl/openssl/blob/d0ff28f8a23f98876a979c932fe1cb016738b0c4/ssl/ssl_cert.c#L866), but in all cases I'm aware of except for one, OpenSSL functions don't take ownership of arguments if the call fails. You have to go source diving do confirm this unfortunately :(.